packages/core/src/StateNode.ts

```diff
@@ -958,11 +958,7 @@ class StateNode<
 
     if (!isInternal) {
       allNextStateNodes.forEach((n1) => {
-        this.nodesFromChild(n1).forEach((n2) => {
-          if (n2 !== this.machine) {
-            reentryNodes.push(n2);
-          }
-        });
```

Comment:
Instead of "patching" the result of `nodesFromChild` I've reexamined the `nodesFromChild`. It was only used here (same for `escapes` that was used in it) so I took the liberty to just remove it (it was a private method) and reintroduce a different implementation, better suited for this particular task.